storm-core/src/jvm/org/apache/storm/cluster/StormClusterStateImpl.java

```diff
@@ -219,6 +219,8 @@ public void stateChanged(CuratorFramework curatorFramework, ConnectionState conn
                 LOG.info("Connection state listener invoked, zookeeper connection state has changed to {}", connectionState);
                 if (connectionState.equals(ConnectionState.RECONNECTED)) {
                     LOG.info("Connection state has changed to reconnected so setting nimbuses entry one more time");
+                    // explicit delete for ephmeral node to ensure this session creates the entry.
+                    stateStorage.delete_node(ClusterUtils.nimbusPath(nimbusId));
```

Comment:
Aren't we using ephemeral node for this? If not I think we should do that no? Instead of manually deleting the node.